### PR TITLE
Move GitHub Actions runner script installation to the golden image

### LINKED ISSUE #602: Move GitHub Actions runner script installation to golden image
Currently, we download it after the vm provisioning. 

https://github.com/ubicloud/ubicloud/blob/9e5228d14954a96cb219b0c9b2ce55e184f9ee08/prog/vm/github_runner.rb#L125

We can download it while generating the golden image.

### PR DESCRIPTION
**Pull request by enescakir**
We setup a vm that boots from the GitHub Actions golden image as GitHub Actions runner after provisioning. This setup includes the installation of the GitHub Actions runner script. The golden image doesn't include the runner script because it's used by Azure Pipelines as well, and GitHub aims to generate platform-agnostic images.
I moved installing the runner script step to the golden image generation process. This change would save approximately 5 seconds when starting a new runner. Additionally, fetching the script directly from GitHub at runtime might be probably a bigger overall weak point.

While creating the golden image, we remove all users at the end of the process. As a result, I've placed the script in the "/usr/local/share/" directory. However, it needs to be moved to the home directory because the runner creates some configuration files at the script location. The "runner" user doesn't have write permission for the "/usr/local/share/" directory.

I have also moved the step for fixing the DNS issue to the golden image generation process.

Fixes #602

### CODE REVIEW

**Review by byucesoy — APPROVED** [review]

File: prog/vm/github_runner.rb [hunk 1]
```diff
@@ -93,28 +93,21 @@ def before_run
     # We need to reconnect machine to load environment variables again.
     vm.sshable.invalidate_cache_entry
 
+    # We placed the script in the "/usr/local/share/" directory while generation
```
Comment (byucesoy) [inline prog/vm/github_runner.rb:96]: while **generating** the golden image

File: prog/vm/github_runner.rb [hunk 2]
```diff
@@ -93,28 +93,21 @@ def before_run
...
+    # the golden image. However, it needs to be moved to a home directory because
```
Comment (byucesoy) [inline prog/vm/github_runner.rb:97]: to **the** home directory

File: prog/vm/github_runner.rb [hunk 3]
```diff
@@ -93,28 +93,21 @@ def before_run
...
+    # the runner creates some configuration files at the root level. The "runner"
```
Comment (byucesoy) [inline prog/vm/github_runner.rb:98]: I didn't understand where does runner creates configuration files. Did you mean at the working directory?

Comment (enescakir) [reply prog/vm/github_runner.rb]: Yeah. It creates at the script location ("/home/runner/actions-runner").

**Pull request merged**